src/java/org/apache/cassandra/cql3/statements/schema/CreateIndexStatement.java

```diff
@@ -93,6 +93,10 @@ public void validate(QueryState state)
     {
         super.validate(state);
 
+        if (!state.getClientState().isInternal && indexName.length() > SchemaConstants.INDEX_NAME_LENGTH)
```

Comment:
IIRC the file names are the result of the concatenation of the keyspace and tablename and the indexname, why don't we consider tableName.length() + keyspaceName.length() here ?

Reply:
No, file names for indexes do not use keyspace and table names when an index name is explicitly given. See [the discussion in the issue](https://github.com/riptano/cndb/issues/13299#issuecomment-2721064053).